common/src/main/java/com/wynntils/features/user/map/MapFeature.java

```diff
@@ -131,6 +136,15 @@ private void openMainMap() {
         McUtils.mc().setScreen(MainMapScreen.create());
     }
 
+    private void openWaypointSetup() {
+        var location = new PoiLocation(
+                McUtils.player().getBlockX(),
+                McUtils.player().getBlockY(),
+                McUtils.player().getBlockZ());
+
+        McUtils.mc().setScreen(PoiCreationScreen.create(null, location));
```

Comment:
I would like you to add a null-check in `PoiCreationScreen#onClose` to handle passing `null` as the old screen value.

Reply:
A null check there would make it impossible to leave the waypoint screen, it has to pass through to the setScreen call

Reply:
Right. I completely forgot that setScreen(null) means to close the screen.. This is why you don't do code reviews after a whole day of competitions..